### DISCUSSION TITLE
What is the use of entityId in while evaluate feature.

### DISCUSSION THREAD

**Discussion by Bhautik0110** [flipt-io/flipt#1276] [Q&A] [answered]
### Question
- What is the use of `entityId` for evaluating features?

```md
HTTP POST /api/v1/evaluate
{
  "requestId": "string",
  "flagKey": "string",
  "entityId": "string",  // what is the use?
  "context": {
    "additionalProp1": "string",
    "additionalProp2": "string",
    "additionalProp3": "string"
  }
}
```

**Comment by markphelps** [comment, marked as answer]
👋🏻 Hi @Bhautik0110 !

`entityId` is used to uniquely id the 'what' you are evaluating so that you always get the same evaluation result for that entity for consistency

https://www.flipt.io/docs/concepts#entities probably explains it better than that

`entityId` could represent a user, IP address, etc. 

Hope that helps!

**Reply by Bhautik0110** [reply]
Gotcha, Thanks @markphelps.

**Reply by Bhautik0110** [reply]
@markphelps please correct me If I'm wrong, At the moment flipt dashboard doesn't have any authentication mechanism provide but API provides it.

**Reply by markphelps** [reply]
@Bhautik0110 thats correct, however we are justt about to release the newest version (likely early next week) which will have authentication for the dashboard/UI as well via [openID connect](https://openid.net/connect/).

This means you'll be able to login using any provider that supports OpenID Connect such as Google, Gitlab, Auth0, Okta, etc!

Following this release we'll work on adding GitHub login as an option as well, since they unfortunately do not support OIDC yet
